drivers/pulse_counter/pulse_counter.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2017 UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     drivers_pulse_counter
+ * @{
+ *
+ * @file
+ * @brief       Driver for the PULSE COUNTER.
+ *
+ * @author      Hyung-Sin Kim <hs.kim@cs.berkeley.edu>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "pulse_counter_params.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define PULSE_COUNTER_NUM    (sizeof(pulse_counter_params)/sizeof(pulse_counter_params[0]))
+
+uint16_t pulse_count[PULSE_COUNTER_NUM];
```

Comment:
I do not really like having this extra `pulse_count` structure here instead of in `auto_init_pulse_counter.c`

Having an indirection is still necessary tough as `saul read` and `saul write` use a `const pulse_counter_t*` pointer, but it could be a pointer and not a table index.

Reply:
Good idea.
Also, why is the pulse count variable defined as a 16 bit uint? It does limit the use cases a bit by having the counter roll over at 65535 counts.
I think either 32 bit or using `unsigned` as the type makes more sense here.

Reply:
@gebart, does SAUL provide 32-bit output? I set this to 16 bits since I thought SAUL provides this type.

Reply:
ah, you are correct. The phydat_t type uses 16 bit ints

Reply:
ok :)

Reply:
I will take the idea of using pointers. Thanks!